library/src/test/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapterBaseTest.java

```diff
@@ -18,22 +18,27 @@
 
 import android.support.v7.widget.RecyclerView;
 import android.view.ViewGroup;
+import com.karumi.headerrecyclerview.testBaseClass.PowermockTest;
 import java.util.List;
 import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
 
 import static junit.framework.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
 
 /**
  * Base test class created to be extended by every HeaderRecyclerViewAdapter test case. Any
  * HeaderRecyclerViewAdapter extension have to match the contract described in this test.
  */
+@PrepareForTest(RecyclerView.ViewHolder.class)
 public abstract class HeaderRecyclerViewAdapterBaseTest<VH extends RecyclerView.ViewHolder, H, T, F>
-    extends RobolectricTest {
+    extends PowermockTest {
```

Comment:
we don't use powermock, because people can write ugly test using it. We prefer use or RobolectricTest if the class that we can test is from the framework, or new android test runner that all methods are not final and you can mock it.